Developers/Unity Project/Spooksville/Assets/Scripts/Items/Weapon.cs

```diff
@@ -2,16 +2,33 @@
 using System.Collections.Generic;
 using UnityEngine;
 
-public abstract class Weapon
+public class Weapon
 {
-    public Sprite sprite;
+    public string Name { get; private set; }
+    public int Damage { get; private set; }
+    public int ID { get; private set; }
+    
+    public Weapon(string name, int damage)
+    {
+        Name = name;
+        Damage = damage;
+
+        SetID();
 
-    public string name;
-    public int damage;
+        WeaponManager.instance.RegisterWeapon(this);
+    }
 
-    public virtual void Initialize()
+    private void SetID()
     {
-        name = null;
-        damage = 1;
+        ID = new System.Random().Next(int.MaxValue);
```

Comment:
What's this ID needed for?
Normally for ID systems, if I need an ID I would bake one into the data so it doesn't change.
If all you need is a unique value, the name should already be unique, right?
You could also make a human- and machine-readable separate ID that makes it obvious it's not intended to be displayed to the player, so `Candy Cane` could have the ID `candy-cane` etc.
Things might get confusing during debugging if each weapon's ID changes every time you run the game.